samples/nrf9160/modem_shell/src/link/link_shell.c

```diff
@@ -1682,6 +1682,387 @@ int link_shell(const struct shell *shell, size_t argc, char **argv)
 	/* Reset getopt for another users: */
 	optreset = 1;
 
-	link_shell_print_usage(&link_cmd_args);
+	link_shell_print_usage(link_cmd_args.command);
 	return ret;
 }
+
+int link_connect(const struct shell *shell, size_t argc, char **argv)
+{
+	int ret = 0;
+	char *apn = NULL;
+	char *family = NULL;
+	char *ip_address = NULL;
+	int protocol = 0;
+	bool protocol_given = false;
+	char *username = NULL;
+	char *password = NULL;
+
+	if (argc < 2) {
+		goto show_usage;
+	}
+	/* Reset getopt due to possible previous failures: */
+	optreset = 1;
+
+	/* We start from subcmd arguments */
+	optind = 1;
+
+	int long_index = 0;
+	int opt;
+	int apn_len = 0;
+
+	mosh_warn("argc=%d, %s, %s, %s", argc, argv[0], argv[1], argv[2]);
+
+	while ((opt = getopt_long(argc, argv,
+				  "a:f:A:P:U:",
+				  long_options, &long_index)) != -1) {
+		switch (opt) {
+
+		case 'a': /* APN */
+			apn_len = strlen(optarg);
+			if (apn_len > LINK_APN_STR_MAX_LENGTH) {
+				mosh_error(
+					"APN string length %d exceeded. Maximum is %d.",
+					apn_len, LINK_APN_STR_MAX_LENGTH);
+				return -EINVAL;
+			}
+			apn = optarg;
+			break;
+		case 'f': /* Address family */
+			family = optarg;
+			break;
+		case 'i': /* IP address */
+			ip_address = optarg;
+			break;
+		case 'A': /* auth protocol */
+			protocol = atoi(optarg);
```

Comment:
Nitpicking: generally it's better to use strtol() instead of atoi() but probably doesn't make any difference here.

Reply:
I'm not planning to do any functional changes within this PR, also to make its review easier. This is supposed to be only a structural change of the subcommands.